common/src/main/java/com/spredfast/kafka/connect/s3/Configure.java

```diff
@@ -65,6 +66,37 @@ public static Converter buildConverter(Map<String, ?> props, String classNamePro
 		}
 	}
 
+	public static HeaderConverter buildHeaderConverter(Map<String, ?> props, String classNameProp, Class<? extends HeaderConverter> defaultConverterClass) {
+		String className = (String) props.get(classNameProp);
+
+		try {
+			HeaderConverter converter;
+
+			if (className == null) {
+				if (defaultConverterClass == null) {
+					return null;
+				} else {
+					converter = defaultConverterClass.newInstance();
+				}
+			} else {
+				converter = (HeaderConverter) Class.forName(className).newInstance();
+			}
+
+			if (converter instanceof Configurable) {
+				((Configurable) converter).configure(props);
+			}
```

Comment:
`converter` is always a `HeaderConverter` (unless the cast on line 82 fails) which is configurable.
```suggestion
		    converter.configure(props);
```